ADR/0036-trusted-artifacts.md

```diff
@@ -0,0 +1,54 @@
+# 36. Trusted Artifacts
+
+Date: 2024-06-11
+
+## Status
+
+Accepted
+
+## Context
+
+One of the properties of Konflux is that users should be allowed to include their own Tekton Tasks
+in a build Pipeline, e.g. to execute unit tests, without jeopardizing the integrity of the build
+process. This is distinct from other build systems where a rigid process prevents users from
+applying such customizations. To support this, Konflux build Pipelines use Trusted Artifacts to
+securely share files between Tasks. Enterprise Contract is then responsible for verifying that
+Trusted Artifacts were properly used in the parts of the build Pipeline that affect the build
+outcome, typically the `git-clone`, `prefetch-dependencies`, and `buildah` Tasks.
+
+Trusted Artifacts is inspired by the upcoming work being done by the Tekton Community,
+[TEP-0139](https://github.com/tektoncd/community/blob/main/teps/0139-trusted-artifacts.md). The
+Konflux version is meant to be a stop-gap until that feature is implemented and ready to be used.
+When the time comes, the Konflux implementation should align with what is provided by the Tekton
+Community and this ADR may need a revision.
+
+In brief, the processes of *creating* a Trusted Artifact wraps files into an archive. Then, the
+location of the archive and its checksum digests are recorded as a Task result. The process of
+*consuming* a Trusted Artifact extracts such an archive, while verifying its checksum digest, into a
+volume only accessible to the Task, e.g. `emptyDir`. The name and the checksum digest of the archive
+is provided via Task parameters. This ensures the artifacts produced by one Task are not tampered
+with when they are consumed by other Tasks.
+
+Furthermore, Konflux takes the approach of sharing such artifacts between Tasks via an OCI registry,
+e.g. quay.io, instead of using Tekton Workspaces backed by Persistent Volume Claims. This has
+several advantages that were previously discussed
+[here](https://github.com/konflux-ci/build-definitions/pull/913#issue-2215784386).
+
+## Decision
+
+Sharing files between Tasks is done via Trusted Artifacts backed by OCI storage.
```

Comment:
Do we want to state that we cannot use the referrer's api here because these archives are generated before we produce an image? I don't know if anyone might wonder about it. (no is a fine response).

Reply:
I think that's too low level for this doc.